src/VisionModel/resnet.jl

```diff
@@ -0,0 +1,72 @@
+using Flux
+
+basicblock(inplanes::Int, outplanes::Int, downsample::Bool = false) = downsample ? 
+  Chain(Conv((3, 3), inplanes => outplanes, stride = 1, pad = 1),
+        BatchNorm(outplanes, λ = relu),
+        Conv((3, 3), outplanes => outplanes, stride = 2, pad = 1,
+        BatchNorm(outplanes, λ=relu)) : 
+  Chain(Conv((3, 3), inplanes => outplanes, stride = 1, pad = 1),
+        BatchNorm(outplanes, λ = relu),
+        Conv((3, 3), outplanes => outplanes, stride = 1, pad = 1,
+        BatchNorm(outplanes, λ=relu))
+
+bottleneck(inplanes::Int, outplanes::Int, downsample::Bool = false) = downsample ?
+  Chain(Conv((1, 1), inplanes => inplanes, stride = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((3, 3), inplanes => inplanes, stride = 1, pad = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((1, 1), inplanes => outplanes, stride = 2),
+        BatchNorm(outplanes, λ = relu)) :
+  Chain(Conv((1, 1), inplanes => inplanes, stride = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((3, 3), inplanes => inplanes, stride = 1, pad = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((1, 1), inplanes => outplanes, stride = 1),
+        BatchNorm(outplanes, λ = relu))
+
+projection(inplanes::Int, outplanes::Int, stride::Int) = Chain(Conv((1, 1), inplanes => outplanes, stride=stride),
+                                                               BatchNorm((outplanes), λ=relu))
+
+identity(inplanes::Int, outplanes::Int, stride::Int) = +
+
+function resnet(block, shortcut, channel_config, block_config)
+  layers = []
+  push!(layers, Conv((7, 7), 3=>inplanes, stride=(2, 2), pad=(3, 3)))
+  push!(layers, BatchNorm(inplanes, λ=relu))
+  push!(layers, MaxPool((3, 3), stride=(2, 2), pad=(1, 1)))
+  inplanes = 64
+  for nrepeats in block_config
+    for i in 1:nrepeats-1
+      for expansion in channel_config
+        outplanes = inplanes * expansion
+        push!(layers, SkipConnection(block(inplanes, outplanes, i == 1),
+                                     identity(inplanes, outplanes, stride)))
+      end
+    end 
+    inplanes *= 2
+    push!(layers, SkipConnection(block(inplanes, outplanes, i == 1),
+                                 projection(inplanes, outplanes, stride)))
+  end
```

Comment:
As per the changes to `basicblock` and `bottleneck`, we don't need to loop over `channel_config` here. But there is still the issue of the very first bottleneck of the first repetition receiving 64 channels instead of 256. To deal with this, I suggest we introduce the concept of `baseplanes`:
```julia
inplanes = 64
baseplanes = 64
for nrepeats in block_config
  outplanes = baseplanes .* channel_config
  # downsample the first block
  if shortcut_config == :A
    # zero padded identity shortcut
  elseif shortcut_config == :B || shortcut_config == :C
    push!(layers, SkipConnection(block(inplanes, outplanes, true),
                                 projection(inplanes, outplanes[end], 2)))
  end
  inplanes = outplanes[end]
  for i in 1:nrepeats
    if shortcut_config == :A || shortcut_config == :B
      push!(layers, SkipConnection(block(inplanes, outplanes, false),
                                   identity(inplanes, outplanes[end], 2)))
    elseif shortcut_config == :C
      # projection shortcut
    end
    inplanes = outplanes[end]
  end
  baseplanes *= 2
end
```